### PR TITLE
sensor cleanups

### PR DESCRIPTION
**Pull request by kkourt**
Please see commits

### CODE REVIEW

File: cmd/tetragon/main.go [hunk 1]
```diff
@@ -246,24 +239,32 @@ func tetragonExecute() error {
 		go crd.WatchTracePolicy(ctx, observer.SensorManager)
 	}
 
-	var startSensors []*sensors.Sensor
+	// load base sensor
+	if err := base.GetInitialSensor().Load(ctx, observerDir, observerDir, option.Config.CiliumDir); err != nil {
```
Comment (olsajiri) [inline cmd/tetragon/main.go:243]: I'm scratching my head how this get compiled.. base.GetInitialSensor returns *sensors.Sensor and we return it as 'err' as error return value, what do I miss?

Comment (kkourt) [reply cmd/tetragon/main.go:243]: `GetInitialSensor()` returns a `*sensors.Sensor`, which defines a `Load(...)` method that returns an error. So doing `GetInitialSensor().Load()` returns an error.

Comment (olsajiri) [reply cmd/tetragon/main.go:243]: aaargh there's .Load .. sorry for noise ;-)

File: pkg/observer/observer.go [hunk 2]
```diff
@@ -107,84 +89,7 @@ func (k *Observer) receiveEvent(data []byte, cpu int) {
...
-	}
-	defer e.CloseAll()
-
-	k.__loopEvents(stopCtx, e)
-	return nil
-}
-
-func (k *Observer) runEventsNew(stopCtx context.Context, ready func()) error {
-	/* Probe runtime configuration and do not fail on errors */
-	k.UpdateRuntimeConf()
```
Comment (kevsecurity) [inline pkg/observer/observer.go:187]: It looks like this line (added to support Cgroups on older kernels AFAIK) has been lost in the process of swapping over.

Comment (willfindlay) [reply pkg/observer/observer.go:187]: Yeah we will need to keep this in, IIRC this is part of what resolved that nasty bug on 4.19 with cgroupsv2.

**Review by willfindlay — APPROVED** [review]
LGTM once @kevsecurity's comment is addressed.

File: pkg/observer/observer.go (hunk 2 above)
Comment (kkourt) [reply pkg/observer/observer.go:187]: ah, nice catch. I messed up the rebase. Let me fix it.

Comment (kkourt) [reply pkg/observer/observer.go:187]: should be OK now, PTAL.

**Review by tixxdz — APPROVED** [review]

**Review by kevsecurity — APPROVED** [review]

**Pull request merged**